### PR TITLE
New endpoint to return a thin list of the applications (as Facets)

### PR DESCRIPTION
**Pull request by josejulio**

### CODE REVIEW

**Comment by burmanm** [comment]
This isn't very descriptive, could you open the reasoning for this or what the FilterOptions means? As in, why is it important in the URL and with label and value ? To me as a user it doesn't tell much (or is there going to be more types under /filterOptions/ ? Why as part of the URL?)

**Comment by josejulio** [comment]
Is a way to get the options that you have to filter by certain parameter. In this case you would do something like:

`GET /api/notifications/v1.0/notifications/filterOptions/applications` and you would get:
```
[
   { label: 'Policies', value: 'the-uuid-for-policies' },
   { label: 'Advisor', value: 'the-uuid-for-advisor' }
]
```

This is used to build the application filter:

Screenshot_2020-11-05_08-15-13


> Why as part of the URL?)

I'm not sure I understand this part. The `value` is passed on a later stage as a `queryParam`, but not on this point.

**Comment by josejulio** [comment]
> or is there going to be more types under /filterOptions

I suppose the same can be done for the other filters: recipients (rbac groups)  and actions(integrations, emails and notifications)

**Comment by burmanm** [comment]
Ouh, you mean like faceting (just without count).

**Comment by josejulio** [comment]
No idea what faceting means in this context.
I was originally going to use the `applications` API to get all the applications and build the filters, but since that's not going to be available to users I ended up making one tailored for the filters.

**Comment by burmanm** [comment]
After thinking a bit, I'd add a count to the possible values also. So one could display at the end of the filtering the count of available  items (if such is chosen to be there). Also, the filterOptions could be "facets" or faceting as that's the usually known term for these types of things. 

The API should be such that one could do it later in the Postgres level if necessary (if performance becomes an issue, otherwise the approach to transform / collectItems / grouping etc is fine).

File: src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java [hunk 1]
```diff
@@ -148,4 +149,10 @@ private String getAddress(Principal principal) {
         return resources.deleteEndpointFromDefaults(principal.getAccount(), endpointId)
                 .onItem().transform(ignored -> Response.ok().build());
     }
+
+    @GET
+    @Path("/facets/applications")
+    public Multi<ApplicationFacet> getApplicationsFacets(@Context SecurityContext sec) {
+        return apps.getApplications().onItem().transform(a -> new ApplicationFacet(a.getName(), a.getId().toString()));
```
Comment (pilhuhn) [inline src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:156]: I think this needs for now the `.collectItems().asList().await().indefinitely();` magic like in `EndpointService.getEndpoints)`

**Review by pilhuhn — CHANGES_REQUESTED** [review]

File: src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java (hunk 1 above)
Comment (burmanm) [reply src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:156]: It seems I can't stop the notifications, but .. don't use ``.await().indefinititely()`` as that's blocking behavior. The use of ``.collectItems().asList()`` is enough and then returning ``Uni<List<ApplicationFacet>>``

.await() should never be used outside tests.

Comment (pilhuhn) [reply src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:156]: Well, apparently the `[][valid data][]` issue we had can be solved by this until a better solution is available.

Comment (theute) [reply src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:156]: @burmanm hey stranger :-) Yes, .await.indefinitely() is the workaround to the "double" result issue we had when you left :) 
There is a proper fix now https://github.com/quarkusio/quarkus/issues/13317 , but we need to upgrade Quarkus and rollback the workaround.
PR are welcome ! :-P

Comment (burmanm) [reply src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:156]: PR is easy, but 1.10.1 universe-bom isn't available in maven-repo yet (even when all the other sub modules are).

https://gist.github.com/burmanm/ce371a45b78b290e77763dd5c0dfbd62

**Comment by pilhuhn** [comment]
On 26 Nov 2020, at 17:20, Michael Burman wrote:

> PR is easy, but 1.10.1 universe-bom isn't available in maven-repo

Is there an additional fix in .1 for this issue? I understand that 
1.10.0 already has the fix

**Comment by burmanm** [comment]
I think 1.10.1 is due to a mistake in publishing (which is why 1.10.0 isn't in the maven repo) and not otherwise different from 1.10.0.

**Review by pilhuhn — APPROVED** [review]

**Comment by pilhuhn** [comment]
Master is now on Quarkus 1.10.1, so we can merge as is.

**Pull request merged**